Source/WebCore/rendering/svg/legacy/LegacyRenderSVGShape.cpp

```diff
@@ -294,7 +294,9 @@ void LegacyRenderSVGShape::paint(PaintInfo& paintInfo, const LayoutPoint&)
             if (svgStyle->shapeRendering() == ShapeRendering::CrispEdges)
                 childPaintInfo.context().setShouldAntialias(false);
 
+            m_fillRequiresClip = !renderingContext.pathClippingIsEntirelyWithinRendererContents();
             fillStrokeMarkers(childPaintInfo);
```

Comment:
Would seem slightly preferable to pass the result of `!renderingContext.pathClippingIsEntirelyWithinRendererContents();` to `fillStrokeMarkers` and `fillShape` instead of adding a state variable, which is reset before this call and then after the call. Reasoning is state can get stale, but really doesn't matter too much since it is only used here.

Reply:
I thought of that but it's several more layers of plumbing, and only applies to the Rect class.

Reply:
Sounds good to leave as-is then 👍